src/db_adapters/receipts.rs

```diff
@@ -112,8 +122,28 @@ async fn find_tx_hashes_for_receipts(
     strict_mode: bool,
     block_hash: &near_indexer::near_primitives::hash::CryptoHash,
     chunk_hash: &near_indexer::near_primitives::hash::CryptoHash,
+    receipts_cache: crate::ReceiptsCache,
 ) -> anyhow::Result<HashMap<String, String>> {
+    let receipts_cache_lock = receipts_cache.lock().await;
+
     let mut tx_hashes_for_receipts: HashMap<String, String> = HashMap::new();
+    //
+    tx_hashes_for_receipts.extend(
+        receipts_cache_lock
+            .iter()
+            .map(|(k, v)| (k.clone(), v.clone())),
+    );
+
+    // discard the Receipts already in cache from the attempts to search
+    receipts.retain(|r| !receipts_cache_lock.contains_key(r.receipt_id.to_string().as_str()));
```

Comment:
Instead of cloning the whole cache, move the retained elements (`retain` returns the removed elements and you don't need to clone them since it returns owned objects)

Reply:
Sorry, I didn't get what you propose